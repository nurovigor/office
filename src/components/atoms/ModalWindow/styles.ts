import { css } from 'linaria';
import { colors } from 'src/components/_settings/_variables';

export const Overlay = css`
	 {
		position: fixed;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		z-index: 1000;
		background-color: ${colors.blackColor};
		opacity: 0.1;
	}
`;

export const Wrapper = css`
	 {
		position: fixed;
		background-color: ${colors.whiteSmokeColor};
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		padding: 50px 50px 25px 50px;
		z-index: 1000;
		min-width: 100px;
		min-height: 50px;
		border-radius: 4px;
	}
`;

export const HeaderModal = css`
	 {
		display: flex;
		position: relative;
		margin-bottom: 10px;
	}
`;

export const Title = css`
	 {
		font-size: 22px;
		font-weight: bold;
		margin: 0 auto;
	}
`;

export const CloseBtn = css`
	 {
		position: absolute;
		right: 0;
		opacity: 0.3;
		cursor: pointer;
		width: 20px;
		height: 20px;

		&:hover {
			opacity: 1;
		}

		&:before,
		&:after {
			position: absolute;
			right: -110%;
			top: -155%;
			content: ' ';
			height: 20px;
			width: 2px;
			background-color: ${colors.darkCharcoal};
		}

		&:before {
			transform: rotate(45deg);
		}

		&:after {
			transform: rotate(-45deg);
		}
	}
`;

export const Content = css`
	 {
		position: relative;
		z-index: 2;
		padding: 10px;
	}
`;
