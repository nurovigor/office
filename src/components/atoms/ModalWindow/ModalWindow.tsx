import React, { ReactNode } from 'react';
import { createPortal } from 'react-dom';
import { Overlay, Wrapper, HeaderModal, CloseBtn, Content, Title } from './styles';

type ModalPropsType = {
	title?: string;
	closeModal: () => void;
	children: ReactNode;
};

let portal = document.getElementById('portal');

if (!portal) {
	portal = document.createElement('div');
	portal.setAttribute('id', 'portal');
	document.body.appendChild(portal);
}

export const ModalWindow: React.FC<ModalPropsType> = ({ title, closeModal, children }) => {
	const onCloseHandle = () => {
		closeModal();
	};

	/*useEffect(() => {
		if (isShow) {
			document.body.style.overflow = 'hidden';
		}
		return () => {
			document.body.style.overflow = 'unset';
		};
	});*/

	return createPortal(
		<>
			<div className={Overlay} onClick={onCloseHandle} />
			<div className={Wrapper}>
				<div className={HeaderModal}>
					<div className={Title}>{title}</div>
					<div data-testid="close-modal" className={CloseBtn} onClick={onCloseHandle} />
				</div>
				<div data-testid="portal" className={Content}>
					{children}
				</div>
			</div>
		</>,
		portal as HTMLElement
	);
};
