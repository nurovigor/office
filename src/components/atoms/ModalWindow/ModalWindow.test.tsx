import React from 'react';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { ModalWindow } from 'src/components/atoms/ModalWindow/ModalWindow';

describe('ModalWindow component', () => {
	let onClose: any;
	beforeEach(() => {
		onClose = jest.fn();
	});
	afterEach(cleanup);
	it('renders ModalWindow', () => {
		render(
			<ModalWindow closeModal={() => {}}>
				<h1>Portal</h1>
			</ModalWindow>
		);
		expect(screen.getByTestId('portal')).toMatchSnapshot();
	});
	it('close modal', () => {
		render(
			<ModalWindow closeModal={onClose}>
				<h1>Portal</h1>
			</ModalWindow>
		);
		fireEvent.click(screen.getByTestId('close-modal'));
		expect(onClose).toBeCalled();
	});
});
