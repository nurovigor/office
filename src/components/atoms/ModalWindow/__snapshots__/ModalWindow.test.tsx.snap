// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow component renders ModalWindow 1`] = `
<div
  data-testid="portal"
>
  <h1>
    Portal
  </h1>
</div>
`;
